lib/stats.rb

```diff
@@ -315,5 +315,35 @@ def goal_diffs
     goal_diffs
   end
 
-  ##== TEAM HELPERS ==##
+  def most_goals_scored_totals
+    team_highest_goals = Hash.new { |hash, key| hash[key] = 0 } # a hash of {team_id: [highest goals scored]}
+
+    @game_teams_data.each do |game_team|
+      team_id = game_team[:team_id]
+      goals_scored = game_team[:goals].to_i
+
+      if goals_scored > team_highest_goals[team_id]
+        team_highest_goals[team_id] = goals_scored
+      end
+    end
+
+  team_highest_goals
+end
+
+def least_goals_scored_totals
```

Comment:
Nice work on these, I wonder if there is a way to condense these into one method